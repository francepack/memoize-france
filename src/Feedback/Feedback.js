import React from 'react';

function Feedback(props) {
  return(
    <div>
    { props.showFeedback &&
      <div className="feedback-overlay">
        <article className="feedback-box">
          { props.questionFeedback &&
          <div> 
            <p className="correct-feedback feedback-head">Correct!</p>
            <p className="sub-feedback">Nailed it - On to the next!</p>
          </div>  
          } { !props.questionFeedback &&
          <div>  
            <p className="incorrect-feedback feedback-head">Sorry - Incorrect!</p>
            <p className="sub-feedback">Check out the below link for review. If you visit this quiz later, see the review section on the bottom of the page to try this one again.</p>
          </div>  
          }
          <p className="learn-more">Learn more at <a href={props.questionInfo}>MDN</a></p>
          <button className="next-btn" onClick={props.toggleFeedback}>Return</button>
        </article>  
      </div>
    }
    </div>
  )
}

export default Feedback;