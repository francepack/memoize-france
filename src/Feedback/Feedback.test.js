import React from 'react';
import Feedback from './Feedback';
import { shallow } from 'enzyme';

import {configure} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });
// const mockQuestion= {id:'1', category:'People', question:'Who is it?', options:['me', 'you', 'neither', 'both'], answer:'you', link:'http://plcaeholder.com'};
const toggleShowFeedback= jest.fn();

describe('Feedback', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Feedback 
        toggleFeedback={toggleShowFeedback}
      />
    )
  });

  it('should match snapshot', () => {
    wrapper.setProps({ showFeedback: true, questionFeedback: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot', () => {
    wrapper.setProps({ showFeedback: true, questionFeedback: false });
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke toggleShowFeedback when the return button is clicked', () => {
    wrapper.setProps({ showFeedback: true, questionFeedback: true });
    wrapper.find('.next-btn').simulate('click', { preventDefault: () => {}});
    expect(toggleShowFeedback).toBeCalled(); 
  });

});  