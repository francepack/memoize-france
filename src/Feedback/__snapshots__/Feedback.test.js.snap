// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback should match snapshot 1`] = `
<div>
  <div
    className="feedback-overlay"
  >
    <article
      className="feedback-box"
    >
      <div>
        <p
          className="correct-feedback"
        >
          Correct!
        </p>
        <p
          className="sub-feedback"
        >
          Nailed it - On to the next!
        </p>
      </div>
       
      <p
        className="learn-more"
      >
        Learn more 
        <a>
          At MDN
        </a>
      </p>
      <button
        className="next-btn"
        onClick={[MockFunction]}
      >
        Return
      </button>
    </article>
  </div>
</div>
`;

exports[`Feedback should match snapshot 2`] = `
<div>
  <div
    className="feedback-overlay"
  >
    <article
      className="feedback-box"
    >
       
      <div>
        <p
          className="incorrect-feedback"
        >
          Sorry - Incorrect!
        </p>
        <p
          className="sub-feedback"
        >
          Check out the below link for review. If you visit this quiz later, see the review section on the bottom of the page to try this one again.
        </p>
      </div>
      <p
        className="learn-more"
      >
        Learn more 
        <a>
          At MDN
        </a>
      </p>
      <button
        className="next-btn"
        onClick={[MockFunction]}
      >
        Return
      </button>
    </article>
  </div>
</div>
`;
